engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala

```diff
@@ -21,15 +29,29 @@ trait KafkaAvroBaseTransformer[T, Y] extends SingleInputGenericNodeTransformatio
 
   override type State = Nothing
 
-  protected def topicParam: Parameter = Parameter[String](KafkaAvroFactory.TopicParamName).copy(editor = Some(FixedValuesParameterEditor(
-    schemaRegistryClient.getAllTopics.getOrElse(Nil).sorted.map(v => FixedExpressionValue(s"'$v'", v))
-  )))
+  protected def topicParam(implicit nodeId: NodeId): WithError[Parameter] = {
+    val topics = schemaRegistryClient.getAllTopics
```

Comment:
I looked to CachedConfluentSchemaRegistryClient and I think we should:

* Change topicsCache to cache, this should be some grouping cache
* At getAllTopics move fixedKey to const and set value `all-topics`

Reply:
I know it's not part of your changes, but maybe you can refactor alsow this? :)

Reply:
refactored a bit :)